src/plugins/intel_gpu/thirdparty/CMakeLists.txt

```
@@ -83,6 +83,14 @@ if(ENABLE_ONEDNN_FOR_GPU)
         set(onednn_gpu_lib "${CMAKE_STATIC_LIBRARY_PREFIX}${DNNL_GPU_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}")
         set(ONEDNN_GPU_LIB_PATH ${ONEDNN_INSTALL_DIR}/lib/${onednn_gpu_lib} CACHE FILEPATH "Path to oneDNN GPU library")
 
+        # remove CMAKE_COMPILE_WARNING_AS_ERROR for onednn_gpu
+        set(_NEW_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+        set(_NEW_CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+        if(WIN32 AND CMAKE_COMPILE_WARNING_AS_ERROR AND CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_VERSION VERSION_LESS 3.24)
+            string(REPLACE "/WX" "" _NEW_CMAKE_CXX_FLAGS "${_NEW_CMAKE_CXX_FLAGS}")
```

```suggestion
            string(REPLACE "/WX" "" CMAKE_CXX_FLAGS "$CMAKE_CXX_FLAGS}")
```
we don't need new variables, because we already inside cmake's `function` with its own scope

---

updated.